File: prisma/refs/java/reorder-linked-list.java
class Solution {
    public void reorderList(ListNode head) {
        if (head == null || head.next == null) return;
        ListNode slow = head, fast = head;
        while (fast.next != null && fast.next.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }
        ListNode second = slow.next;
        slow.next = null;
        ListNode prev = null;
        while (second != null) {
            ListNode nx = second.next;
            second.next = prev;
            prev = second;
            second = nx;
        }
        ListNode first = head;
        while (prev != null) {
            ListNode fn = first.next, rn = prev.next;
            first.next = prev;
            prev.next = fn;
            first = fn;
            prev = rn;
        }
    }
}

File: prisma/refs/py/reorder-linked-list.py
def reorder_list(head):
    if not head or not head.next:
        return
    slow, fast = head, head
    while fast.next and fast.next.next:
        slow = slow.next
        fast = fast.next.next
    second, slow.next = slow.next, None
    prev = None
    while second:
        second.next, prev, second = prev, second, second.next
    first = head
    while prev:
        first.next, prev.next, first, prev = prev, first.next, first.next, prev.next
